src/wui/game_message_menu.cc

```diff
@@ -487,14 +487,7 @@ void GameMessageMenu::filter_messages(Widelands::Message::Type const msgtype) {
 		toggle_filter_messages_button(*scenariobtn_, msgtype);
 		break;
 
-	case Widelands::Message::Type::kNoMessages:
-	case Widelands::Message::Type::kAllMessages:
-	case Widelands::Message::Type::kGameLogic:
-	case Widelands::Message::Type::kEconomySiteOccupied:
-	case Widelands::Message::Type::kWarfareSiteDefeated:
-	case Widelands::Message::Type::kWarfareSiteLost:
-	case Widelands::Message::Type::kWarfareUnderAttack:
-	case Widelands::Message::Type::kTradeOfferReceived:
+	default:
```

Comment:
I believe the case list was intended to ensure that only our defined messge types are processed. Other types would throw an error by falling through to default, which is defined as unreachabele code.  So why to skip this safeguarded list. (